libraries/AP_HAL/utility/RingBuffer.h

```diff
@@ -110,6 +112,16 @@ class ObjectBuffer {
         delete buffer;
     }
 
+    // return size of ringbuffer
+    uint32_t get_size(void) const { return buffer->get_size() / sizeof(T); }
+
+    // set size of ringbuffer, caller responsible for locking
+    bool set_size(uint32_t size) { return buffer->set_size(((size+1) * sizeof(T))); }
+
+    // read len bytes without advancing the read pointer
```

Comment:
why does this return in units of bytes? Wouldn't it make more sense to be in units of number of vars of type T ?

Reply:
yes!

Reply:
Fixed